std/range/package.d

```diff
@@ -7980,6 +7980,13 @@ sorting relation.
         import std.algorithm.iteration : chunkBy;
         return _input.chunkBy!((a, b) => !predFun(a, b) && !predFun(b, a));
     }
+
+    import std.algorithm.sorting: MultiPred;
+
+    static if( is( pred == MultiPred!P, P... ) )
+    auto shrinkPred( uint n )() {
```

Comment:
Think it would be nicer to have this as a freestanding UFCS function. Then you can also put it next to multiSort in std/algorithm/sorting.d.
You should be able to match `shrinkPred(less...)(SortedRange!(MultiPred!less) sr)`.